src/classes/ACCT_CannotDelete_TDTM.cls

```diff
@@ -34,67 +34,78 @@
                                 (SELECT ID FROM Account.Course_Enrollments__r LIMIT 1),
+                                (SELECT ID FROM Account.Education_History__r LIMIT 1),
                                 (SELECT ID FROM Account.Facilities__r LIMIT 1),
                                 (SELECT ID FROM Account.Household_Members__r LIMIT 1),
                                 (SELECT ID FROM Account.Organization_Members__r LIMIT 1),
                                 (SELECT ID FROM Account.Program_Enrollments__r LIMIT 1), 
                                 (SELECT ID FROM Account.Program_Plans__r LIMIT 1),
                                 (SELECT ID FROM Account.Terms__r LIMIT 1),
                                 (SELECT ID FROM Account.Time_Blocks__r LIMIT 1)
-                                FROM Account WHERE ID in :oldlist]) {
-                
-                if (hasChildRecords(a)) {
-                    Account accountInContext = oldMap.get(a.ID);
+                                FROM Account WHERE ID IN :oldlist]) {
+
+                if (hasChildRecords(acc)) {
+                    Account accountInContext = oldMap.get(acc.ID);
                     accountInContext.addError(Label.CannotDelete);
                 }
             }     
         }
         return new DmlWrapper();
     }
-    
+
     /*******************************************************************************************************
      * @description Evaluates whether the Account has any child related records.
      * @param a is the current Account record.
      * @return Boolean.
      ********************************************************************************************************/
-    private static Boolean hasChildRecords(Account a) {
-        return (a.Addresses__r.size() > 0 || a.Affl_Contacts__r.size() > 0 || a.Attributes__r.size() > 0 ||
-            a.Courses__r.size() > 0 || a.Course_Enrollments__r.size() > 0 || a.Facilities__r.size() > 0 ||
-            a.Household_Members__r.size() > 0 || a.Organization_Members__r.size() > 0 || a.Program_Enrollments__r.size() > 0 || 
-            a.Program_Plans__r.size() > 0 || a.Terms__r.size() > 0 || a.Time_Blocks__r.size() > 0); 
+    private static Boolean hasChildRecords(Account acc) {
+        return (acc.Addresses__r.size() > 0 ||
+                acc.Affl_Contacts__r.size() > 0 ||
+                acc.Attributes__r.size() > 0 ||
+                acc.Courses__r.size() > 0 ||
+                acc.Course_Enrollments__r.size() > 0 ||
+                acc.Education_History__r.size() > 0 ||
+                acc.Facilities__r.size() > 0 ||
+                acc.Household_Members__r.size() > 0 ||
+                acc.Organization_Members__r.size() > 0 ||
+                acc.Program_Enrollments__r.size() > 0 ||
+                acc.Program_Plans__r.size() > 0 ||
+                acc.Terms__r.size() > 0 ||
+                acc.Time_Blocks__r.size() > 0);
```

Comment:
We should be using isEmpty(). Splitting this out across each line made this much easier to read.
```suggestion
        return (acc.Addresses__r.isEmpty() == false ||
                acc.Affl_Contacts__r.isEmpty() == false ||
                acc.Attributes__r.isEmpty() == false ||
                acc.Courses__r.isEmpty() == false ||
                acc.Course_Enrollments__r.isEmpty() == false ||
                acc.Education_History__r.isEmpty() == false ||
                acc.Facilities__r.isEmpty() == false ||
                acc.Household_Members__r.isEmpty() == false ||
                acc.Organization_Members__r.isEmpty() == false ||
                acc.Program_Enrollments__r.isEmpty() == false ||
                acc.Program_Plans__r.isEmpty() == false ||
                acc.Terms__r.isEmpty() == false ||
                acc.Time_Blocks__r.isEmpty() == false);
```